src/backend/common/CommandBuffer.cpp

```diff
@@ -472,6 +594,28 @@ namespace backend {
                             return false;
                         }
 
+                        // TODO(kainino@chromium.org): We should be able to
+                        // optimize this by tracking which textures are in use
+                        // as attachments. Maybe it's possible that the
+                        // XAttachment usages could mark this: disallow
```

Comment:
I was thinking of disallowing explicit transitions INTO XAttachment, but having render passes always leave their attachments in the 'no usage' state (if they're not frozen). Then we don't have to set the end usage in the render pass.